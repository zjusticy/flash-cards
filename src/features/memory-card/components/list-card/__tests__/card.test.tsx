import React from 'react';
// import renderer from "react-test-renderer";
// import configureStore from 'redux-mock-store'
// import { shallow } from 'enzyme';
// import toJson from 'enzyme-to-json';
import { beforeEach, describe, it, expect } from 'vitest';

import { render, cleanup } from '@testing-library/react';

import Card from '../list-card';

beforeEach(() => cleanup);

describe('Card', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <Card active index={0} title="Title" front="hello" clicked={() => {}} />
    );
    // expect(getByRole("button").classList.contains("Big")).toBe(true);
    // expect(getByRole("button").textContent).toEqual("Facebook");

    expect(asFragment()).toMatchSnapshot();
  });
});
