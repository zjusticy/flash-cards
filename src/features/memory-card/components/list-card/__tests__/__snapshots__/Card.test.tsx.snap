// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="card bg-[#FBE091]"
    role="button"
    tabindex="0"
  >
    <h3>
      Title
    </h3>
    <p>
      hello
    </p>
  </div>
</DocumentFragment>
`;

exports[`Card renders correctly 1`] = `
<DocumentFragment>
  <div
    class="card active"
    role="button"
    tabindex="0"
  >
    <h3>
      Title
    </h3>
    <p>
      hello
    </p>
  </div>
</DocumentFragment>
`;
